### DISCUSSION TITLE
Deploy with Docker on Openshift (kubernetes)

### DISCUSSION THREAD

**Discussion by sendaops** [nextgenhealthcare/connect#5493] [General]
Hi, I'm trying to get this work on Openshift (kubernetes) directly from Dockerfile, this is my deployment config and the error I'm getting when pod is started. Any suggestion? Thanks!

**_ERROR >>_** sed: couldn't open temporary file /opt/connect/conf/sedFzQel1: Permission denied

_**DeploymentConfig:**_ see attach
[dc.txt](https://github.com/nextgenhealthcare/connect/files/9852335/dc.txt)

**Comment by jonbartels** [comment]
https://mirthconnect.slack.com/archives/CJKSCKQMR/p1631028489032700 is a user with a similar problem.

**Reply by jonbartels** [reply]
You can see where `sed` is being called at https://github.com/nextgenhealthcare/connect-docker/search?q=sed&type=code 

I ran into this with my devops team for our k8s based deployments. What `sed` does is that it makes it's changes using a temp file, then swaps that out for the target file if there are no errors. You have to ensure that sed has permissions to both create that temp file and overwrite the target file. 

IIRC we solved it by ensuring the right user was being used when `entrypoint.sh` executes but I do not know exactly.

**Reply by sendaops** [reply]
Hi, the thing is, I can't change anything because I'm using as a source "image: docker.io/nextgenhealthcare/connect"
I think I must set a user permission in my deploymentconfig to avoid that error, but I can't find the proper way to do it.
